test/e2e/testnet/node.go

```diff
@@ -221,11 +221,15 @@ func (n *Node) Init(genesis *types.GenesisDoc, peers []string, configOptions ...
 		return fmt.Errorf("writing address book: %w", err)
 	}
 
-	if err := n.Instance.AddFolder(nodeDir, remoteRootDir, "10001:10001"); err != nil {
-		return fmt.Errorf("copying over node %s directory: %w", n.Name, err)
+	err = n.Instance.Commit()
+	if err != nil {
+		return fmt.Errorf("committing instance: %w", err)
 	}
 
-	return n.Instance.Commit()
+	if err = n.Instance.AddFolder(nodeDir, remoteRootDir, "10001:10001"); err != nil {
+		return fmt.Errorf("copying over node %s directory: %w", n.Name, err)
+	}
```

Comment:
So we are supposed to add the folder after committing now?

Reply:
The instances do not differ except for the files we add there.
- If we add the files before committing, it will build a container for each Instance, which takes very long if we want to start 100 validators
- If we add the files after committing, knuu can use a cached version of the Instance, which speeds up if you want to start many nodes that use the same base, but different files.

Reply:
what's the best way to keep track of important details such as this @smuu? we could add a code comment above, but I imagine this would also be important to know when creating tests for node as well

Reply:
I've created a tracking issue for this https://github.com/celestiaorg/celestia-app/issues/3601